src/status_im/transport/inbox.cljs

```diff
@@ -136,34 +137,46 @@
-        wnode    (get-current-wnode-address db)]
+  (let [web3          (:web3 db)
+        wnode         (get-current-wnode-address db)
+        peers-summary (:peers-summary db)
+        connected?    (registered-peer? peers-summary wnode)]
     (when config/offline-inbox-enabled?
-      (handlers-macro/merge-fx cofx
-                               {::add-peer {:wnode wnode}
-                                :utils/dispatch-later [{:ms connection-timeout
-                                                        :dispatch [:inbox/connection-check]}]}
-                               (update-mailserver-status :connecting)
-                               (generate-mailserver-symkey)))))
+      (if connected?
+        (handlers-macro/merge-fx cofx
+                                 (update-mailserver-status :connected)
+                                 (generate-mailserver-symkey))
+        (handlers-macro/merge-fx cofx
+                                 {::add-peer {:wnode wnode}
+                                  :utils/dispatch-later [{:ms connection-timeout
+                                                          :dispatch [:inbox/connection-check]}]}
+                                 (update-mailserver-status :connecting)
+                                 (generate-mailserver-symkey))))))
 
 (defn peers-summary-change-fx
-  "Called when a peer summary signal is received or we want to try to connect
-   to mailserver ie. after login.
-   Marks the mailserver as trusted peer if it is already added to the list of
-   peers or connects to it if it's not the case"
-  [{:keys [db] :as cofx}]
+  "There is only 2 summary changes that require offline inboxing action:
+  - mailserver disconnected: we try to reconnect
+  - mailserver connected: we mark the mailserver as trusted peer"
+  [previous-summary {:keys [db] :as cofx}]
   (when (and (:account/account db)
              config/offline-inbox-enabled?)
-    (let [{:keys [peers-summary peers-count]} db]
-      (if (zero? peers-count)
-        (update-mailserver-status :disconnected cofx)
-        (let [wnode (get-current-wnode-address db)]
-          (if (registered-peer? peers-summary wnode)
-            (handlers-macro/merge-fx cofx
-                                     {::mark-trusted-peer {:web3  (:web3 db)
-                                                           :wnode wnode}}
-                                     (generate-mailserver-symkey))
-            (connect-to-mailserver cofx)))))))
+    (let [{:keys [peers-summary peers-count]} db
+          wnode                               (get-current-wnode-address db)
+          mailserver-was-registered?          (registered-peer? previous-summary
+                                                                wnode)
+          mailserver-is-registered?           (registered-peer? peers-summary
+                                                                wnode)
+          mailserver-connected?               (and mailserver-is-registered?
```

Comment:
Just to make sure I get this straight:

Here we only want to run `mark-trustued-peer` when we go from `unregistered`->`registered` ?

We don't run it when we go from `registered`->`registered`, which makes sense. 

If that is the case, I would maybe change the name of the variable `mailserver-connected?` as if I am in case 2 (`registered->registered`) I would expect it also to be `true`.

Not sure exactly what a good name is, random thoughts: `unregistered->registered?`, `has-been-registered?`, `moved-to-registered?`, basically the idea is to capture a `transition` to a different `state` (we can drop `mailserver` as probably does not provide any extra info).

But not a huge issue.

Reply:
Maybe disconnected->connected? and connected->disconnected?

Reply:
but I already "hear" the state transition in "mailserver-connected?" to me it implies it wasn't connected before and just connected ? Is it a wrong assomption ?

Reply:
The first time I looked at the code I thought you were only checking if it was `connected`, not whether it transition to connected. But I take people might think differently. (also `->connected?` could be an option). but up to you, minor point.

Reply:
there is definitely an ambiguity with "connected" because it is meant to be the verb at the past tense but can be interpretted as the adjective describing the state.